tests/cpgrid/grid_lgr_test.cpp

```diff
@@ -470,17 +470,93 @@ void refinePatch_and_check(Dune::CpGrid& coarse_grid,
         //
         const auto& leaf_view = coarse_grid.leafGridView();
         const auto& level0_view = coarse_grid.levelGridView(0);
+
         Dune::MultipleCodimMultipleGeomTypeMapper<Dune::CpGrid::LeafGridView> leafMapper(leaf_view, Dune::mcmgElementLayout());
         Dune::MultipleCodimMultipleGeomTypeMapper<Dune::CpGrid::LevelGridView> level0Mapper(level0_view, Dune::mcmgElementLayout());
+
         const auto& leaf_idSet = (*data[startIJK_vec.size()+1]).local_id_set_;
+        const auto& leaf_globalIdSet = (*data[startIJK_vec.size()+1]).global_id_set_;
```

Comment:
Maybe do it the DUNE way and ask the grid for the id set (`grid.globalIdSet())?

Reply:
Same as before, I use localIdSet() and (new method) globalIdSet() for each level grid, or the leaf grid view (CpGridData objects)